dhis-2/dhis-api/src/main/java/org/hisp/dhis/organisationunit/OrganisationUnit.java

```diff
@@ -605,6 +607,24 @@ public void updateParent( OrganisationUnit newParent )
         newParent.getChildren().add( this );
     }
 
+    /**
+     * Indicates whether this org unit is a descendant of any of the given
+     * ancestor org units.
+     *
+     * @param ancestors the collection of ancestor org units.
+     * @return true if his org unit is a descendant of the ancestors.
+     */
+    public boolean isDescendant( Collection<OrganisationUnit> ancestors )
+    {
+        if ( isEmpty( ancestors ) )
+        {
+            return false;
+        }
+
+        return IdentifiableObjectUtils.getUidsAsSet( ancestors ).stream()
+            .anyMatch( uid -> StringUtils.contains( path, uid ) );
```

Comment:
I think it is better to use 
```java
return ancestors.stream().map(OrganisationUnit::getUid).anyMatch(uid -> path.contains(uid));
```
This avoids building the set of UIDs and as `path` and `uid` should be non-null we can use `contains` directly.

Or if we assume this might be called with non persisted OUs
```java
return ancestors.stream()
  .map(OrganisationUnit::getUid)
  .filter(Objects::nonNull)
  .anyMatch(uid -> path.contains(uid));
```
This would highlight how this situation is handled. Also something we might want to mention in the javadoc.

Reply:
Agreed, I had refactored some code in the "previous" isDescendant method and missed that we create two streams here, good catch.